extensions/functions/src/test/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregationTest.java

```diff
@@ -154,15 +155,20 @@ public void test_random_type_random_values() throws Exception {
             .collect(Collectors.toList());
         var type = RandomPicks.randomFrom(RandomizedContext.current().getRandom(), validTypes);
 
-        var data = TestingHelpers.getRandomsOfType(10000, 10000, type)
+        // For IP type we must ensure that provided values are handled similar to IpColumnReference.
+        Function ipFormatter = null;
+        if(type.id() == DataTypes.IP.id()) {
+            ipFormatter = o -> NetworkAddress.format(InetAddresses.forString((String) o));
+        }
+
+        var data = TestingHelpers.getRandomsOfType(10000, 10000, type, ipFormatter)
```

Comment:
Instead of adding a workaround here we could adjust the data generation function to only return formatted values that match the `IpColumnReference` output: 

https://github.com/crate/crate/blob/6c57ce984602071fdadc16233732625ce5f0e9ec/server/src/testFixtures/java/io/crate/testing/DataTypeTesting.java#L93-L100

Reply:
Done. I was thinking about doing so at the beginning but had concern that some other tests using that generator might not expected specially formatted ip - but as long as tests are passing it's not a problem.